### ISSUE TITLE
Compilation of gpt-fast fails with `torch._dynamo.exc.Unsupported: UNPACK_SEQUENCE MapVariable()`

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#132044]
### 🐛 Describe the bug

After another update, compilation of `gpt-fast` failed for me with 
```
  File "/Users/malfet/git/pytorch/pytorch/torch/_dynamo/symbolic_convert.py", line 1880, in UNPACK_SEQUENCE
    unimplemented(f"UNPACK_SEQUENCE {seq}")
  File "/Users/malfet/git/pytorch/pytorch/torch/_dynamo/exc.py", line 271, in unimplemented
    raise Unsupported(msg, case_name=case_name)
torch._dynamo.exc.Unsupported: UNPACK_SEQUENCE MapVariable()

from user code:
   File "/Users/malfet/git/pytorch-labs/gpt-fast/generate.py", line 65, in decode_one_token
    logits = model(x, input_pos)
  File "/Users/malfet/git/pytorch/pytorch/torch/nn/modules/module.py", line 1746, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/malfet/git/pytorch-labs/gpt-fast/model.py", line 130, in forward
    x = layer(x, input_pos, freqs_cis, mask)
  File "/Users/malfet/git/pytorch/pytorch/torch/nn/modules/module.py", line 1746, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/malfet/git/pytorch-labs/gpt-fast/model.py", line 149, in forward
    h = x + self.attention(self.attention_norm(x), freqs_cis, mask, input_pos)
  File "/Users/malfet/git/pytorch/pytorch/torch/nn/modules/module.py", line 1746, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/malfet/git/pytorch-labs/gpt-fast/model.py", line 191, in forward
    q, k, v = map(lambda x: x.transpose(1, 2), (q, k, v))
```

while compiling following code snippet:
```python
    def forward(self, x: Tensor, freqs_cis: Tensor, mask: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
        bsz, seqlen, _ = x.shape

        kv_size = self.n_local_heads * self.head_dim
        q, k, v = self.wqkv(x).split([self.dim, kv_size, kv_size], dim=-1)

        q = q.view(bsz, seqlen, self.n_head, self.head_dim)
        k = k.view(bsz, seqlen, self.n_local_heads, self.head_dim)
        v = v.view(bsz, seqlen, self.n_local_heads, self.head_dim)
    
        q = apply_rotary_emb(q, freqs_cis)
        k = apply_rotary_emb(k, freqs_cis)

        q, k, v = map(lambda x: x.transpose(1, 2), (q, k, v))
    
        if self.kv_cache is not None:
            k, v = self.kv_cache.update(input_pos, k, v)
    
        k = k.repeat_interleave(self.n_head // self.n_local_heads, dim=1)
        v = v.repeat_interleave(self.n_head // self.n_local_heads, dim=1)
        y = F.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0)
    
        y = y.transpose(1, 2).contiguous().view(bsz, seqlen, self.dim)
            
        y = self.wo(y)
        return y
```

### Versions

[trunk](https://github.com/pytorch/pytorch/commit/be3eba382f638f60a084514a464a8f9b7665e186) 

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @amjames

**Comment by malfet** [comment]
Let me know if someone else is working on it, otherwise I'd like to try to fix it tomorrow

**Comment by Skylion007** [comment]
Isn't this related to: https://github.com/pytorch/pytorch/pull/131961

**Comment by joydddd** [comment]
https://github.com/pytorch/pytorch/pull/131961 doesn't fix this issue

**Comment by williamwen42** [comment]
This should fix it https://github.com/pytorch/pytorch/pull/132069

**Comment by joydddd** [comment]
> This should fix it #132069

Confirmed.
